src/test/kotlin/blackjack/model/card/CardDeckTest.kt

```diff
@@ -0,0 +1,99 @@
+package blackjack.model.card
+
+import blackjack.model.strategy.NormalStrategy
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class CardDeckTest {
+    @Test
+    fun `카드 덱에서 카드 한장을 받는다`() {
+        val cardDeck = CardDeck(NormalStrategy().shuffle())
+        val result = cardDeck.get()
+
+        result shouldBe Card(CardNumber.ACE, CardSuit.클럽)
+    }
+
+    @Test
+    fun `카드 덱에서 카드 한장을 받으면 덱의 카드 수가 1 감소한다`() {
+        val cardDeck = CardDeck(NormalStrategy().shuffle())
+        val beforeSize = cardDeck.deck.size
+        cardDeck.get()
+        val afterSize = cardDeck.deck.size
+
+        beforeSize - 1 shouldBe afterSize
+    }
+
+    @Test
+    fun `카드 덱에 카드를 추가하면 덱의 카드 수가 1 증가한다`() {
+        val cardDeck = CardDeck()
+        val beforeSize = cardDeck.deck.size
+        cardDeck.add(Card(CardNumber.ACE, CardSuit.클럽))
+        val afterSize = cardDeck.deck.size
+
+        beforeSize + 1 shouldBe afterSize
+    }
+
+    @Test
+    fun `카드 덱에 카드가 없는 경우 카드를 받을 수 없다`() {
+        val cardDeck = CardDeck()
+
+        assertThrows<IllegalStateException> {
+            cardDeck.get()
+        }
+    }
+
+    @Test
+    fun `카드 덱에 이미 존재하는 카드를 받을 수 없다`() {
+        val cardDeck = CardDeck(NormalStrategy().shuffle())
+
+        assertThrows<IllegalStateException> {
+            cardDeck.add(Card(CardNumber.ACE, CardSuit.클럽))
+        }
+    }
+
+    @Test
+    fun `카드 덱의 합이 21보다 큰 경우 버스트이다`() {
+        val cardDeck = CardDeck()
+        cardDeck.add(Card(CardNumber.KING, CardSuit.클럽))
+        cardDeck.add(Card(CardNumber.KING, CardSuit.다이아몬드))
+        cardDeck.add(Card(CardNumber.KING, CardSuit.스페이드))
```

Comment:
테스트에서 가능한 구현한 내용이 노출되지 않는 것이 좋습니다.

운영코드의 변화가 테스트코드의 변화로 이어질 수 있지만, 가능한 변화로 이어지지 않으면 유지보수하기 더 좋습니다.

```suggestion
                val cardDeck = CardDeck(
            listOf(
                Card(CardNumber.KING, CardSuit.클럽),
                Card(CardNumber.KING, CardSuit.다이아몬드),
                Card(CardNumber.KING, CardSuit.스페이드),
            )
        )
```

Reply:
무조건 add를 해야한다 생각해서 구현했는데 객체 생성시에 리스트로 넣어도 체크가 정상적으로 동작하는군요?!
더 가독성 좋게 수정된다니 너무 좋습니다~!